### PR TITLE
created Debug.PhysicsViewer

### PR DESCRIPTION
**Pull request by abow**
How to use:

http://www.babylonjs-playground.com/#CA51CM

### CODE REVIEW

File: src/Debug/babylon.physicsViewer.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                if(this._imposters[i].isDisposed){
+                    this.hideImposter(this._imposters[i--]);
+                }else{
+                    plugin.syncMeshWithImposter(this._meshes[i], this._imposters[i]);
+                }
+            }
+
+        }
+
+        public showImposter(imposter:PhysicsImpostor):void{
```
Comment (deltakosh) [inline src/Debug/babylon.physicsViewer.ts]: ShowImpostor?

File: src/Debug/babylon.physicsViewer.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                }
+            }
+
+            var debugMesh = this._physicsEnginePlugin.getDebugMesh(imposter, this._scene);
+
+            if(debugMesh){
+                this._imposters[this._numMeshes] = imposter;
+                this._meshes[this._numMeshes] = debugMesh;
+
+                if(this._numMeshes == 0){
```
Comment (deltakosh) [inline src/Debug/babylon.physicsViewer.ts]: ===

File: src/Debug/babylon.physicsViewer.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                    }else{
+                        this._meshes[0] = null;
+                        this._imposters[0] = null;
+                    }
+                    removed = true;
+                    break;
+                }
+            }
+
+            if(removed && this._numMeshes == 0){
```
Comment (deltakosh) [inline src/Debug/babylon.physicsViewer.ts]: ===

File: src/Debug/babylon.physicsViewer.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                this._scene.registerBeforeRender(this._renderFunction);
```
Comment (deltakosh) [inline src/Debug/babylon.physicsViewer.ts]: Unregister instead?

File: src/Debug/babylon.physicsViewer.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+        }
+
+        public hideImposter(imposter:PhysicsImpostor){
+
+            var removed = false;
+
+            for (var i = 0; i < this._numMeshes; i++){
+                if(this._imposters[i] == imposter){
+                    this._scene.removeMesh(this._meshes[i]);
```
Comment (deltakosh) [inline src/Debug/babylon.physicsViewer.ts]: The mesh was already removed I guess

File: src/Physics/Plugins/babylon.cannonJSPlugin.ts [hunk 6]
```diff
@@ -469,8 +472,65 @@
...
+            var body = imposter.physicsBody;
+            var shape = body.shapes[0];
+            var mesh:InstancedMesh;
+            
+            if (shape.halfExtents) {
+                if(!this._debugBoxMesh){
+                    this._debugBoxMesh = MeshBuilder.CreateBox('physicsBodyBoxViewMesh', { size: 1 }, scene); 
+                    this._debugBoxMesh.renderingGroupId = 1;
+                    this._debugBoxMesh.rotationQuaternion = Quaternion.Identity();
+                    this._debugBoxMesh.material = new StandardMaterial('', scene);
```
Comment (deltakosh) [inline src/Physics/Plugins/babylon.cannonJSPlugin.ts]: We should use the same material for all and dispose it and the end

File: src/Physics/Plugins/babylon.oimoJSPlugin.ts [hunk 7]
```diff
@@ -388,7 +391,65 @@ module BABYLON {
             }
         }
 
+        public getDebugMesh(imposter:PhysicsImpostor, scene:Scene):AbstractMesh{
+
```
Comment (deltakosh) [inline src/Physics/Plugins/babylon.oimoJSPlugin.ts]: Same comments apply here ;)

**Comment by abow** [comment]
Thanks for the feedback.   

Here is the new demo:
http://www.babylonjs-playground.com/#CA51CM#1

**Pull request merged**